internal/controllers/topology/cluster/patches/variables/variables.go

```diff
@@ -271,16 +271,19 @@ func ControlPlane(cpTopology *clusterv1.ControlPlaneTopology, cp, cpInfrastructu
 }
 
 // MachineDeployment returns variables that apply to templates belonging to a MachineDeployment.
-func MachineDeployment(mdTopology *clusterv1.MachineDeploymentTopology, md *clusterv1.MachineDeployment, mdBootstrapTemplate, mdInfrastructureMachineTemplate *unstructured.Unstructured) ([]runtimehooksv1.Variable, error) {
+func MachineDeployment(mdTopology *clusterv1.MachineDeploymentTopology, md *clusterv1.MachineDeployment, mdBootstrapTemplate, mdInfrastructureMachineTemplate *unstructured.Unstructured, definitionFrom string, validDefinitions map[string]bool) ([]runtimehooksv1.Variable, error) {
 	variables := []runtimehooksv1.Variable{}
 
 	// Add variables overrides for the MachineDeployment.
 	if mdTopology.Variables != nil {
 		for _, variable := range mdTopology.Variables.Overrides {
-			variables = append(variables, runtimehooksv1.Variable{
-				Name:  variable.Name,
-				Value: variable.Value,
-			})
+			// Add the variable if it is defined for the current patch or it is defined for all the patches.
+			if variable.DefinitionFrom == allPatchesDefinitionFrom || variable.DefinitionFrom == definitionFrom {
+				// only add the variable if it has a valid definition in the ClusterClass.
```

Comment:
Similar comment to the one in engine. "has a valid definition in the ClusterClass" seems misleading as we're not checking for validity we're checking if the variable has been defined for the specific "definitionFrom"